src/components/KMultiselect/KMultiselect.vue

```diff
@@ -416,63 +417,79 @@ const props = defineProps({
     default: false,
   },
   /**
-   * Test mode - for testing only, strips out generated ids
+   * Dropdown footer text
    */
-  testMode: {
-    type: Boolean,
-    default: false,
-  },
-  /**
-  * Dropdown footer text
-  */
   dropdownFooterText: {
     type: String,
     default: '',
   },
   /**
-    * Dropdown footer text position
-    * Accepted values: 'sticky' and 'static'
-    */
+   * Dropdown footer text position
+   * Accepted values: 'sticky' and 'static'
+   */
   dropdownFooterTextPosition: {
     type: String as PropType<DropdownFooterTextPosition>,
     default: 'sticky',
   },
 })
 
-const emit = defineEmits(['selected', 'item:added', 'item:removed', 'input', 'change', 'update:modelValue', 'query-change'])
+const emit = defineEmits<{
+  (e: 'selected', item: MultiselectItem[]): void
+  (e: 'input', value: string[]): void
+  (e: 'change', item: MultiselectItem | null): void
+  (e: 'update:modelValue', value: string[]): void
+  (e: 'query-change', query: string): void
+  (e: 'item-added', value: MultiselectItem): void
+  (e: 'item-removed', value: MultiselectItem): void
+}>()
 
 const isRequired = computed((): boolean => attrs.required !== undefined && String(attrs.required) !== 'false')
 const strippedLabel = computed((): string => stripRequiredLabel(props.label, isRequired.value))
 const hasLabelTooltip = computed((): boolean => !!(props.labelAttributes?.help || props.labelAttributes?.info || slots['label-tooltip']))
+const hasDropdownFooter = computed((): boolean => !!(props.dropdownFooterText || slots['dropdown-footer-text']))
+
+const getBadgeAppearance = (item?: MultiselectItem): BadgeAppearance => {
+  if (isDisabled.value || isReadonly.value || item?.disabled) {
```

Comment:
question: how would `item` be undefined?

Reply:
The badge that's used for +1 (hidden items tooltip)